src/main/core/worker.rs

```diff
@@ -677,8 +677,10 @@ mod export {
     use super::*;
 
     #[no_mangle]
-    pub extern "C-unwind" fn worker_getDNS() -> *mut cshadow::DNS {
-        Worker::with_dns(std::ptr::from_ref).cast_mut()
+    pub extern "C-unwind" fn worker_getHostsFilePath() -> *mut std::ffi::c_char {
+        Worker::with_dns(|dns| unsafe {
+            cshadow::dns_getHostsFilePath(std::ptr::from_ref(dns).cast_mut())
+        })
```

Comment:
This should have a comment about whether the returned string pointer needs to be freed, or how long it lives. (Or ignore if you plan on removing this anyways as part of the rust migration in the follow up MR.)

Reply:
Yes, this is changed in #3464, and the safety requirements are documented there too.